com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateSubscribersStep.java

```diff
@@ -288,11 +281,7 @@ private boolean updateCurrentEnvironment(Map<String, String> currentEnvironment,
     }
 
     private String getFirstComponent(String propertyName) {
-        int index = propertyName.indexOf(ValidatorUtil.DEFAULT_SEPARATOR);
-        if (index != -1) {
-            return propertyName.substring(0, index);
-        }
-        return propertyName;
+        return StringUtils.substringBefore(propertyName, ValidatorUtil.DEFAULT_SEPARATOR);
```

Comment:
This again changes behaviour - the new code will return empty string instead of the whole propertyName. Add test.

Reply:
No, it won't. This is the implementation of the substringBefore method 
`if (isEmpty(str) || separator == null) {
            return str;
        }
        if (separator.isEmpty()) {
            return EMPTY;
        }
        final int pos = str.indexOf(separator);
        if (pos == INDEX_NOT_FOUND) {
            return str;
        }
        return str.substring(0, pos);
`